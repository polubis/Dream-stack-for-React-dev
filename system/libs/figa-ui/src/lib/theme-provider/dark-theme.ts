import type { Theme } from './defs';
import { tokens } from './tokens';

const dark: Theme = {
  outline: {
    color: tokens.common.white,
  },
  switch: {
    bg: tokens.gray[50],
    color: tokens.common.white,
    active: {
      bg: tokens.green[200],
    },
  },
  scroll: {
    thumb: tokens.gray[600],
    track: tokens.gray[550],
    thumbHover: tokens.gray[650],
  },
  divider: {
    default: {
      bg: tokens.common.white,
    },
    primary: {
      bg: tokens.primary[100],
    },
  },
  radio: {
    borderColor: tokens.common.white,
    active: {
      bg: tokens.green[200],
      borderColor: tokens.green[200],
    },
  },
  checkbox: {
    borderColor: tokens.common.white,
    checked: {
      bg: tokens.primary[50],
      borderColor: tokens.primary[50],
      color: tokens.common.white,
    },
  },
  tabs: {
    filled: {
      bg: tokens.gray[500],
      active: {
        borderColor: tokens.green[50],
      },
    },
  },
  chips: {
    color: tokens.common.white,
    borderColor: tokens.common.white,
    active: {
      color: tokens.green[50],
      borderColor: tokens.green[50],
    },
  },
  badge: {
    outlined: {
      primary: {
        color: tokens.primary[50],
        borderColor: tokens.primary[50],
      },
      secondary: {
        color: tokens.primary[100],
        borderColor: tokens.primary[100],
      },
      ok: {
        color: tokens.green[100],
        borderColor: tokens.green[100],
      },
      casual: {
        color: tokens.gray[150],
        borderColor: tokens.gray[150],
      },
    },
    filled: {
      primary: {
        color: tokens.common.black,
        bg: tokens.primary[50],
      },
      secondary: {
        color: tokens.common.black,
        bg: tokens.primary[100],
      },
      ok: {
        color: tokens.common.black,
        bg: tokens.green[100],
      },
      casual: {
        color: tokens.common.black,
        bg: tokens.gray[150],
      },
    },
  },
  thumbnail: {
    bg: tokens.backdrop[100],
    color: tokens.common.white,
  },
  filePicker: {
    outline: {
      color: tokens.primary[50],
    },
    invalid: {
      color: tokens.red[150],
      outline: tokens.red[150],
    },
  },
  field: {
    invalid: {
      color: tokens.red[100],
    },
    hint: {
      color: tokens.common.white,
    },
  },
  font: {
    default: {
      color: tokens.common.white,
    },
    primary: {
      color: tokens.primary[100],
    },
  },
  body: {
    bg: tokens.common.black,
  },
  alert: {
    filled: {
      info: {
        bg: tokens.gray[150],
        color: tokens.common.black,
      },
      ok: {
        bg: tokens.green[150],
        color: tokens.common.black,
      },
      error: {
        bg: tokens.red[50],
        color: tokens.common.white,
      },
      warn: {
        bg: tokens.yellow[50],
        color: tokens.common.black,
      },
    },
    outlined: {
      info: {
        color: tokens.common.white,
      },
      ok: {
        color: tokens.green[150],
      },
      error: {
        color: tokens.red[150],
      },
      warn: {
        color: tokens.yellow[50],
      },
    },
  },
  modal: {
    bg: tokens.common.black,
    border: tokens.dark[50],
    backdrop: tokens.backdrop[50],
  },
  button: {
    filled: {
      primary: {
        color: tokens.common.black,
        bg: tokens.primary[50],
        outlineColor: tokens.common.white,
      },
      secondary: {
        color: tokens.common.black,
        bg: tokens.primary[100],
        outlineColor: tokens.common.white,
      },
      tertiary: {
        color: tokens.common.black,
        bg: tokens.common.white,
        outlineColor: tokens.common.white,
      },
    },
    outlined: {
      primary: {
        color: tokens.primary[50],
        bg: 'transparent',
        borderColor: tokens.primary[50],
        outlineColor: tokens.common.white,
      },
      secondary: {
        color: tokens.primary[100],
        bg: 'transparent',
        borderColor: tokens.primary[100],
        outlineColor: tokens.common.white,
      },
      tertiary: {
        color: tokens.common.white,
        bg: 'transparent',
        borderColor: tokens.common.white,
        outlineColor: tokens.common.white,
      },
    },
    ghost: {
      hoverBg: tokens.dark[250],
    },
  },
  select: {
    bg: tokens.dark[50],
    color: tokens.common.white,
    placeholderColor: tokens.gray[50],
    hoverBg: tokens.gray[100],
    optionSeparator: tokens.gray[100],
    optionActiveColor: tokens.secondary[50],
  },
  nav: {
    bg: tokens.dark[100],
    borderColor: tokens.gray[300],
    bgWithOpacity: tokens.dark[200],
  },
  logo: {
    text: {
      first: tokens.purple[100],
      second: tokens.common.white,
      last: tokens.secondary[200],
    },
    bg: {
      first: {
        from: tokens.red[50],
        to: tokens.primary[250],
      },
      second: {
        from: tokens.secondary[250],
        to: tokens.primary[250],
      },
      last: {
        from: tokens.blue[200],
        to: tokens.primary[250],
      },
    },
  },
  progressCircle: {
    bg: tokens.dark[50],
  },
  input: {
    filled: {
      bg: tokens.dark[50],
      color: tokens.common.white,
      placeholderColor: tokens.gray[50],
      fxColor: tokens.common.white,
    },
    outlined: {
      borderColor: tokens.common.white,
      color: tokens.common.white,
      placeholderColor: tokens.gray[50],
      fxColor: tokens.common.white,
    },
    empty: {
      color: tokens.common.white,
      placeholderColor: tokens.gray[50],
      fxColor: tokens.common.white,
    },
    invalid: tokens.red[150],
  },
  placeholder: {
    filled: {
      bg: tokens.gray[300],
    },
    outlined: {
      bg: tokens.common.black,
      borderColor: tokens.gray[300],
    },
  },
  box: {
    filled: {
      bg: tokens.gray[300],
    },
    outlined: {
      bg: tokens.common.black,
      borderColor: tokens.gray[300],
    },
  },
  footer: {
    bg: tokens.common.black,
    borderColor: tokens.gray[300],
  },
  code: {
    bg: tokens.dark[150],
    gutters: tokens.dark[150],
    selectionBg: '#4d4d4d',
    foldPlaceholder: '#4d4d4d',
    tags: [
      { tag: 'angleBracket', color: '#da6bad' },
      { tag: 'arithmeticOperator', color: '#fff' },
      { tag: 'bitwiseOperator', color: '#fff' },
      { tag: 'blockComment', color: '#6A9955' },
      { tag: 'bool', color: '#569cd6' },
      { tag: 'brace', color: '#ffd70f' },
      { tag: 'bracket', color: '#da6bad' },
      { tag: 'className', color: '#3abfb0' },
      { tag: 'comment', color: '#6A9955' },
      { tag: 'compareOperator', color: '#fff' },
      { tag: 'controlKeyword', color: '#eba0d9' },
      { tag: 'definition', color: '#4fc1ff' },
      { tag: 'definitionKeyword', color: '#3981c5' },
      { tag: 'definitionOperator', color: '#fff' },
      { tag: 'derefOperator', color: '#fff' },
      { tag: 'function', color: '#dcdc9f' },
      { tag: 'keyword', color: '#3981c5' },
      { tag: 'labelName', color: '#4fc1ff' },
      { tag: 'lineComment', color: '#6A9955' },
      { tag: 'literal', color: '#ef8c59' },
      { tag: 'logicOperator', color: '#fff' },
      { tag: 'moduleKeyword', color: '#eba0d9' },
      { tag: 'name', color: '#4fc1ff' },
      { tag: 'null', color: '#569cd6' },
      { tag: 'number', color: '#accea8' },
      { tag: 'operatorKeyword', color: '#3abfb0' },
      { tag: 'paren', color: '#da6bad' },
      { tag: 'typeName', color: '#3abfb0' },
    ],
  },
  codeBlock: {
    header: {
      bg: tokens.gray[350],
      dots: {
        first: {
          bg: tokens.green[50],
        },
        second: {
          bg: tokens.yellow[50],
        },
        third: {
          bg: tokens.blue[650],
        },
      },
    },
    content: {
      bg: tokens.dark[150],
    },
  },
  list: {
    marker: {
      bg: tokens.primary[50],
    },
  },
  blockquote: {
    bg: tokens.gray[450],
    borderColor: tokens.gray[300],
  },
  loader: {
    primary: tokens.common.white,
    secondary: tokens.primary[50],
  },
  avatars: {
    restCounter: {
      bg: tokens.primary[50],
      color: tokens.common.black,
    },
  },
  creatorLayout: {
    borderColor: tokens.gray[300],
    bg: tokens.common.black,
  },
};

export { dark };
