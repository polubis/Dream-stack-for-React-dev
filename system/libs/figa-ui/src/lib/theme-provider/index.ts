export * from './theme-provider';
export * from './defs';
