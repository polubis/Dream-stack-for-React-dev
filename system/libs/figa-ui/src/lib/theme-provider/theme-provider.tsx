import { useState, createContext, useMemo, useContext } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import type {
  ThemeKey,
  ThemeProviderProps,
  ThemeProviderValue,
  ThemesList,
  UnsafeThemeProviderValue,
  GetDefaultValue,
} from './defs';
import { themes } from './themes';
import { GlobalStyle } from './global-style';

const Ctx = createContext<UnsafeThemeProviderValue>(null);

const defaultKey = 'dark' as ThemeKey;

const getDefaultValue: GetDefaultValue = ({
  key = defaultKey,
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  setTheme = () => {},
} = {}) => ({
  key,
  setTheme,
  themes,
  theme: themes[key],
  themesList: Object.entries(themes) as ThemesList,
});

const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const [key, setKey] = useState(defaultKey);

  const value: ThemeProviderValue = useMemo(
    () =>
      getDefaultValue({
        key,
        setTheme: setKey,
      }),
    [key]
  );

  return (
    <Ctx.Provider value={value}>
      <StyledThemeProvider theme={value.theme}>
        <GlobalStyle />
        {children}
      </StyledThemeProvider>
    </Ctx.Provider>
  );
};

const useThemeProvider = (): ThemeProviderValue => {
  const ctx = useContext(Ctx);

  return ctx ? ctx : getDefaultValue();
};

export { ThemeProvider, useThemeProvider };
