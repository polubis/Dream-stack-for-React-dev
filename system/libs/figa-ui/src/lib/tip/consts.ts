const tip_orientations = ['top', 'right', 'bottom', 'left'] as const;

export { tip_orientations };
