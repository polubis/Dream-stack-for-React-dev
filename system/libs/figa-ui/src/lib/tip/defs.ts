import type { ReactNode } from 'react';
import type { SpacingKey } from '../theme-provider';
import type { tip_orientations } from './consts';

type TipOrientation = (typeof tip_orientations)[number];

interface TipProps {
  className?: string;
  children: ReactNode;
  content: ReactNode;
  offset?: SpacingKey;
  orientation?: TipOrientation;
}

export type { TipProps, TipOrientation };
