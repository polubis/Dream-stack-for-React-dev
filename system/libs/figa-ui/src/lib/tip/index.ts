export * from './tip';
export * from './defs';
