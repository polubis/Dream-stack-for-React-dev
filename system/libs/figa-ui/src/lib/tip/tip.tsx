import type { TipProps } from './defs';
import c from 'classnames';
import styled from 'styled-components';
import { tokens } from '../theme-provider';
import { Font } from '../font';
import { row, spacing } from '../shared';

const Container = styled.div<Required<Pick<TipProps, 'offset'>>>`
  position: relative;
  width: max-content;

  &.top {
    .tip-content {
      top: -${(props) => spacing.get(props.offset)};
    }

    &:hover .tip-content {
      transform: translateY(-100%);
    }
  }

  &.right {
    .tip-content {
      top: 0;
      right: ${(props) => `calc(-${spacing.get(props.offset)} * 2)`};
      bottom: 0;
    }

    &:hover .tip-content {
      transform: translateX(100%);
    }
  }

  &.bottom {
    .tip-content {
      top: ${(props) => spacing.get(props.offset)};
    }

    &:hover .tip-content {
      transform: translateY(100%);
    }
  }

  &.left {
    .tip-content {
      top: 0;
      left: ${(props) => `calc(-${spacing.get(props.offset)} * 2)`};
      bottom: 0;
    }

    &:hover .tip-content {
      transform: translateX(-100%);
    }
  }

  &:hover .tip-content {
    opacity: 1;
  }

  .tip-content {
    ${row()}
    height: 100%;
    opacity: 0;
    transform: translateY(0);
    transition: all 0.3s cubic-bezier(0.19, 1, 0.22, 1);
    transition-delay: 0.2s;
    position: absolute;
    padding: ${tokens.spacing[100]};
    background: ${(props) => props.theme.box.filled.bg};
    border-radius: ${tokens.radius[50]};
    z-index: ${tokens.z[800]};
    width: max-content;
  }
`;

const Tip = ({
  className,
  children,
  content,
  offset = 50,
  orientation = 'bottom',
}: TipProps) => {
  return (
    <Container className={c('tip', className, orientation)} offset={offset}>
      {children}
      <div className="tip-content">
        <Font variant="b3">{content}</Font>
      </div>
    </Container>
  );
};

export { Tip };
