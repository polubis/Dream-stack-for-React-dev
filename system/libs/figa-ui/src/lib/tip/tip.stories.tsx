import type { Story, Meta } from '@storybook/react';

import { Tip } from './tip';
import { Box } from '../box';
import { Font } from '../font';

export default {
  component: Tip,
  title: 'Tip',
} as Meta;

const Template: Story = () => {
  return (
    <Box padding={[300, 300, 300, 300]} spacing={[1000, 1000, 1000, 1000, 1000]}>
      <Box padding={[1000, 0, 0, 0]}>
        <Tip content="This is my tip!" orientation="top">
          <Font variant="h5">I'm top and have content</Font>
        </Tip>
      </Box>
      <Box>
        <Tip content="This is my tip!" orientation="right">
          <Font variant="h5">I'm right and have content</Font>
        </Tip>
      </Box>
      <Box>
        <Tip content="This is my tip!">
          <Font variant="h5">I'm bottom and have content</Font>
        </Tip>
      </Box>
      <Box padding={[0, 0, 0, 1000]}>
        <Tip content="This is my tip!" orientation="left">
          <Font variant="h5">I'm left and have content</Font>
        </Tip>
      </Box>
      <Box style={{ height: '100px', overflow: 'scroll' }}>
        <Tip content="This is my tip with custom offset" orientation="bottom">
          <Font variant="h5">I work in scrollable area</Font>
        </Tip>
        <Font variant="h1">
          undoubtable source. Lorem Ipsum comes from sections 1.10.32 and
          1.10.33 of "de Finibus Bonorum et Malorum" (The Extremes of Good and
          Evil) by Cicero, written in 45 BC. This book is a treatise on the
          theory of ethics, very popular during the Renaissance. The first line
          of Lorem Ipsum, "Lorem ipsum dolor sit amet..", comes from a line in
          section 1.10.32. The standard chunk of Lorem Ipsum used since the
          1500s is reproduced below for those interested. Sections 1.10.32 and
          1.10.33 from "de Finibus Bonorum et Malorum" by Cicero are also
          reproduced in their exact original form
        </Font>
      </Box>
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {};
