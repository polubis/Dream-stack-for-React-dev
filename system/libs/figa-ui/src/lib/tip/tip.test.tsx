import { render } from '@testing-library/react';
import { Tip } from './tip';
import { ThemeProvider } from '../theme-provider';

describe('Tip can be used when: ', () => {
  it('[FRAGILE] works by default', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Tip content="My tip here">
          <div>Content</div>
        </Tip>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] assigns classes and orientation', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <Tip
          className="my-class"
          orientation="right"
          offset={150}
          content="My tip here"
        >
          <div>Content</div>
        </Tip>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
