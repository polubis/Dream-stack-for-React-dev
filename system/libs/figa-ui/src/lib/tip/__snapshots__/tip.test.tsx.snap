// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tip can be used when:  [FRAGILE] assigns classes and orientation 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe fXbzPf tip my-class right"
    offset="150"
  >
    <div>
      Content
    </div>
    <div
      class="tip-content"
    >
      <p
        class="font b3 default"
      >
        My tip here
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tip can be used when:  [FRAGILE] works by default 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe jlLCWP tip bottom"
    offset="50"
  >
    <div>
      Content
    </div>
    <div
      class="tip-content"
    >
      <p
        class="font b3 default"
      >
        My tip here
      </p>
    </div>
  </div>
</DocumentFragment>
`;
